code/modules/mob/living/simple_animal/hostile/space_dragon.dm

```diff
@@ -0,0 +1,727 @@
+		QDEL_NULL(src)
+
+/mob/living/simple_animal/hostile/space_dragon/AttackingTarget()
+	if(using_special)
+		return
+	if(target == src)
+		to_chat(src, "<span class='warning'>You almost bite yourself, but then decide against it.</span>")
+		return
+	if(istype(target, /turf/closed/wall))
+		if(tearing_wall)
+			return
+		tearing_wall = TRUE
+		var/turf/closed/wall/thewall = target
+		to_chat(src, "<span class='warning'>You begin tearing through the wall...</span>")
+		playsound(src, 'sound/machines/airlock_alien_prying.ogg', 100, TRUE)
+		var/timetotear = 4 SECONDS
+		if(istype(target, /turf/closed/wall/r_wall))
+			timetotear = 12 SECONDS
+		tearing_wall = TRUE
+		if(do_after(src, timetotear, target = thewall))
+			if(istype(thewall, /turf/open))
+				return
+			thewall.dismantle_wall(1)
+			playsound(src, 'sound/effects/meteorimpact.ogg', 100, TRUE)
+		tearing_wall = FALSE
+		return
+	if(isliving(target)) //Swallows corpses like a snake to regain health.
+		var/mob/living/L = target
+		if(L.stat == DEAD)
+			if(is_swallowing)
+				return
+			to_chat(src, "<span class='warning'>You begin to swallow [L] whole...</span>")
+			is_swallowing = TRUE
+			if(do_after(src, 3 SECONDS, target = L))
+				is_swallowing = FALSE
+				if(eat(L))
+					adjustHealth(-L.maxHealth * 0.25)
+			return
+	. = ..()
+	if(istype(target, /obj/mecha))
+		var/obj/mecha/M = target
+		M.take_damage(50, BRUTE, "melee", 1)
+
+/mob/living/simple_animal/hostile/space_dragon/Move()
+	if(!using_special)
+		..()
+
+/mob/living/simple_animal/hostile/space_dragon/OpenFire()
+	if(using_special)
+		return
+	ranged_cooldown = world.time + ranged_cooldown_time
+	fire_stream()
+
+/mob/living/simple_animal/hostile/space_dragon/death(gibbed)
+	empty_contents()
+	if(!objective_complete)
+		destroy_rifts()
+	..()
+	add_dragon_overlay()
+	UnregisterSignal(small_sprite, COMSIG_ACTION_TRIGGER)
+
+/mob/living/simple_animal/hostile/space_dragon/revive(full_heal, admin_revive)
+	. = ..()
+	add_dragon_overlay()
+	RegisterSignal(small_sprite, COMSIG_ACTION_TRIGGER, .proc/add_dragon_overlay)
+
+/mob/living/simple_animal/hostile/space_dragon/wabbajack_act(mob/living/new_mob)
+	empty_contents()
+	. = ..()
+
+/**
+  * Allows space dragon to choose its own name.
+  *
+  * Prompts the space dragon to choose a name, which it will then apply to itself.
+  * If the name is invalid, will re-prompt the dragon until a proper name is chosen.
+  */
+/mob/living/simple_animal/hostile/space_dragon/proc/dragon_name()
+	var/chosen_name = sanitize_name(reject_bad_text(stripped_input(src, "What would you like your name to be?", "Choose Your Name", real_name, MAX_NAME_LEN)))
+	if(!chosen_name)
+		to_chat(src, "<span class='warning'>Not a valid name, please try again.</span>")
+		dragon_name()
+		return
+	to_chat(src, "<span class='notice'>Your name is now <span class='name'>[chosen_name]</span>, the feared Space Dragon.</span>")
+	fully_replace_character_name(null, chosen_name)
```

Comment:
This is probably a bad idea ngl

Reply:
ask admins, I'll remove if needed

Reply:
we've decided on a pre-selected lists of proper nouns and adjectives for names like
`Dragur the Terrible`. You will be able to pick from both lists so you can have a static but reasonable space dragon name.

Reply:
nevermind, too much work. commented out the line that calls the rename method. Leaving this in though